### ISSUE TITLE
[Accessibility][Usable - Bot Responses-Edit]: No Keys Information is provided for keyboard users to navigate out from the edit field.

### ISSUE THREAD

**Issue by ShellySoni** [microsoft/BotFramework-Composer#2092]
**User Experience:**
For better usability purpose and for keyboard users it will be good if proper information about keys to be used is provided either in terms of tooltip or some pop-up as provided using screen reader.

**Test Environment:**
Environment: Chromium Edge (Anaheim) + Narrator
Build Version: 2004 (OS Build 19569.1000)​​
Microsoft Edge Version: 82.0.436.0 (Official build) canary (64-bit)​
Test URL: http://localhost:3000/home

**Repro Steps:**
1. Open URL: http://localhost:3000/home
2. "Bot Framework Composer" homepage will gets opened.
3. Select any of the previously created bot from "Recent Bots" or create one from the available Bots under 'Examples' and activate it.
4. Note: Better to create new "Dialog Actions" bot from example.
5. Select 'Bot Responses' from left navigation.
6. Navigate throughout the page when "Edit toggle button is enabled".
7. Observe the issue when user reach within the edit field.

**Actual Result:**
No Information is provided i.e. what keys should be used for keyboard users that will help to navigate out from the edit field. Normally 'esc' key is used but here esc key is also not working.

**Note:**
Same issue is repro within 'User Input - Edit Mode'.

**Observation:**
By using screen reader, user is able to know the keys i.e. Press 'Alt+F1' to open the dialog, then use 'Ctrl+M' key to navigate out.

**Expected Result:**
Proper Information should be provided i.e. what keys should be used for keyboard users that will help to navigate out from the edit field. Information either in terms of Tooltip or some pop-up should be provided.

**MAG Guidelines:**
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/Doc.aspx?sourcedoc=%7bF98F4235-B6BD-47D3-8DA4-F28B9B6FBFA8%7d&file=Guideline%20101%20%E2%80%93%20Recommended%20Assistive%20Technology%20Tools%20for%20MAS%20Testing%20of%20Microsoft%20Digital%20Properties.docx&action=default&mobileredirect=true

**Attachment:**
[Usable_Bot Responses_Edit Mode.zip](https://github.com/microsoft/BotFramework-Composer/files/4253682/Usable_Bot.Responses_Edit.Mode.zip)

**Comment by corinagum** [comment]
Can we also get a11y tags added to this issue? Including severity :)

@ShellySoni, is there a resource for seeing out other code editors within Microsoft (for example, VS Code or Visual Studio) handle keyboard flow for accessibility?

**Comment by zhixzhan** [comment]
Should we re-consider this? I know the Esc key is currently used for dismiss suggestions. And it's very easy to press Esc, use click `Edit mode` button to exit seems more reliable.

![Untitled2](https://user-images.githubusercontent.com/49866537/78329258-c7ce4000-75b3-11ea-8039-e08b231057ad.gif)

**Comment by liweitian** [comment]
@Andy I tried on https://github.com/microsoft/BotFramework-Composer/pull/2329. Clicking esc will jump out of edit mode directly. Do we need to reconsider this like zhixiang said?

**Comment by a-b-r-o-w-n** [comment]
That's a great point @zhixzhan. @beyackle didn't you say there is already a key combo that removes focus from the editor?

**Comment by beyackle** [comment]
Ctrl-M, at least in Edge and Firefox, switches between "use Tab to move to the next element" and "use Tab to insert tabs" mode, and the screen-reader follows along with this focus.

**Comment by a-b-r-o-w-n** [comment]
Is there any indication to a SR or KB user to do this? If not, maybe we need some documentation added.

**Comment by beyackle** [comment]
It wouldn't hurt. I don't know what documentation screen-readers typically come with or how much training people who need to use them get, so they might not be aware that it's an option. Also we should check more browsers to see how standard a key shortcut this is.

**Comment by a-b-r-o-w-n** [comment]
Yeah. I'm thinking since monaco is basically its own application hosted inside of ours, it would probably be helpful to have a pointer to documentation somewhere.

It could be a KB/SR only visible link before the editors. @sangwoohaan @mareekuh what do you think?

**Comment by beyackle** [comment]
Oh, that explains why I couldn't find that Ctrl-M thing in the usual docs. It's a Monaco thing. So, yes, we should definitely document that somewhere.

**Comment by corinagum** [comment]
@awalia13, @ShellySoni, Please see the following gif: 

![Screen Recording 2020-04-08 at 10 20 AM](https://user-images.githubusercontent.com/3619060/78813944-ae960c80-7982-11ea-936b-92e6336947ab.gif)

The Monaco editor has documentation on accessibility. In this case I don't think a tooltip is appropriate, so would it be sufficient to provide a documentation pointer (as described by @a-b-r-o-w-n above) in Composer?

**Comment by mareekuh** [comment]
"[..] It could be a KB/SR only visible link before the editors. @sangwoohaan @mareekuh what do you think?[..]"

@a-b-r-o-w-n - yes, I like that suggestion.
@zxyanliu - would you be able to assist with this?

**Comment by zxyanliu** [comment]
@mareekuh 
I would be happy to help. I was not aware of this keyboard usability issue and it seems to be important to some users. 
If we have the keyboard shortcut combo, I would be more than happy to add them to the current [Composer documentation](https://docs.microsoft.com/en-us/composer/). Please let me know.

**Comment by ashish315** [comment]
@corinagum I am agree with the idea of documentation pointer. If the KB user will aware how to exit from the edit field or the shortcut key, it is helpful for them.

**Comment by mareekuh** [comment]
Is this being worked on? Is the keiboardshortcut cheat-sheet available in the documentation?
@zxyanliu

**Comment by beyackle** [comment]
This is definitely a documentation issue, and a cheat-sheet would be a good way to solve it.

**Comment by zxyanliu** [comment]
Sure. I can add a keyboard short cut cheat-sheet in the Composer docs. Who should I contact for the list of the short cut?

**Comment by beyackle** [comment]
@zxyanliu I found this, which looks like it'd be a good basis for one: https://github.com/Microsoft/monaco-editor/wiki/Monaco-Editor-Accessibility-Guide

**Comment by Amit8527510735** [comment]
#Nonbenchmark;
